tests/unit/OmiseTest.php

```diff
@@ -1,37 +1,121 @@
+            )
+            ->getMock();
+
+        $this->smarty = $this->getMockBuilder(stdClass::class)
+            ->setMockClassName('Smarty')
+            ->setMethods(
+                array(
+                    'assign',
+                )
+            )
             ->getMock();
 
         $this->omise = new Omise();
+        $this->omise->setSetting($this->setting);
+        $this->omise->setSmarty($this->smarty);
+    }
+
+    public function testConstructor_whenNewTheInstance_theDefaultValueOfTheAttributeSettingMustBeAvailable()
+    {
+        $omise = new Omise();
+
+        $setting = $omise->getSetting();
+
+        $this->assertNotEmpty($setting);
     }
 
-    public function testName_omise()
+    public function testName_theNameThatUsedToReferenceInTheProgramMustBe_omise()
     {
         $this->assertEquals('omise', $this->omise->name);
     }
 
-    public function testDisplayName_Omise()
+    public function testDisplayName_theNameThatUsedToDisplayeToTheMerchantMustBe_Omise()
     {
         $this->assertEquals('Omise', $this->omise->displayName);
     }
 
-    public function testNeedInstance_0()
+    public function testNeedInstance_noNeedToLoadModuleAtTheBackendModulePage()
     {
         $this->assertEquals(0, $this->omise->need_instance);
     }
 
-    public function testBootstrap_true()
+    public function testBootstrap_bootstrapTemplateIsRequired()
     {
         $this->assertEquals(true, $this->omise->bootstrap);
     }
+
+    public function testGetContent_merchantOpenTheSettingPage_retrieveSettingDataFromTheDatabaseAndDisplayOnThePage()
+    {
+        $this->setting->method('getLivePublicKey')->willReturn('live_public_key');
+        $this->setting->method('getLiveSecretKey')->willReturn('live_secret_key');
+        $this->setting->method('isModuleEnabled')->willReturn('module_status');
+        $this->setting->method('isSandboxEnabled')->willReturn('sandbox_status');
+        $this->setting->method('getSubmitAction')->willReturn('submit_action');
+        $this->setting->method('getTestPublicKey')->willReturn('test_public_key');
+        $this->setting->method('getTestSecretKey')->willReturn('test_secret_key');
+        $this->setting->method('getTitle')->willReturn('title');
+
+        $this->smarty->expects($this->exactly(8))
+            ->method('assign')
+            ->withConsecutive(
+                array('live_public_key', 'live_public_key'),
+                array('live_secret_key', 'live_secret_key'),
+                array('module_status', 'module_status'),
+                array('sandbox_status', 'sandbox_status'),
+                array('submit_action', 'submit_action'),
+                array('test_public_key', 'test_public_key'),
+                array('test_secret_key', 'test_secret_key'),
+                array('title', 'title')
+            );
+
+        $this->omise->getContent();
+    }
+
+    public function testGetContent_merchantSaveSetting_theSettingDataMustBeSaved()
```

Comment:
@guzzilar 

> But it's not wrong to double change

It is double developers effort.

> which is you expected that the save() method from Setting will be triggered once and that's it, right?

No, I expect the setting data must be saved.

> but the Setting::save() method can't really save the data to the DB (maybe, somehow we forget to execute Configuration::updateValue() function)

It is out of scope of a class, `Omise`. The scope of a unit test case `OmiseTest` is a class `Omise`. It does not need to care another class or dependency.

Each unit test class will test a base class and ensure the base class is correctly work.

> So, the statement theSettingDataMustBeSaved will be false but the test will pass.

With unit test case it do not really know and do not care that another class will pass or fail because all dependencies have been mocked.

---

I still prefer a benefit of unit test case that can has more value in part of description more than just for test a method has been executed.

This benefit can be given to other developers or contributors. Who do not have the knowledge or have the little knowledge about the system to be able more understand.

For this time, you can change the expectation part to what you understand and I will review it.